py/desispec/scripts/tile_redshifts.py

```diff
@@ -404,11 +405,12 @@ def write_redshift_script(batchscript, outdir,
         echo $(basename $spectra) already exists, skipping grouping
     else
         # Check if any input frames exist
-        CFRAMES=$(ls {frame_glob})
-        MISSING_CFRAMES=$?
+        # Use either .fits or .fits.gz search will fail and throw error, so catch them
+        CFRAMES=$(ls {frame_glob} 2>/dev/null)
+        NUM_TERMS=$(echo "{frame_glob}" | wc -w)
         NUM_CFRAMES=$(echo $CFRAMES | wc -w)
-        if [ $MISSING_CFRAMES -ne 0 ] && [ $NUM_CFRAMES -gt 0 ]; then
-            echo ERROR: some expected cframes missing for spectrograph $SPECTRO but proceeding anyway
+        if [ $NUM_TERMS -gt $NUM_CFRAMES ]; then
+            echo ERROR: some expected cframes missing for spectrograph $SPECTRO with $NUM_TERMS search terms but only $NUM_CFRAMES found. Proceeding anyway
```

Comment:
After writing and deleting 3 draft comments, I'll discuss this with you in person.

Reply:
As we discussed I have explicitly defined the number of exposures rather than having bash count globs. I replaced the ERROR message here with an informative message used for all spectrographs (not under an `if` statement). I did retain a WARNING statement that just helps guide the eye (or a `grep`) if the number of `cframes` is fewer than the number of exposures. That can genuinely happen since the code isn't smart enough to figure out how many it should have, but I think a WARNING is useful even if there are false alarms. Let me know if you disagree and we can iterate.